spikeinterface/core/segmentutils.py

```diff
@@ -314,6 +314,185 @@ def get_unit_spike_train(self, *args, **kwargs):
+
+        # Check and pull n_samples from each segment
+        parent_segments = []
+        parent_num_samples = []
+        for sorting_i, sorting in enumerate(sorting_list):
+            for segment_i, parent_segment in enumerate(sorting._sorting_segments):
+                # Check t_start is not assigned
+                segment_t_start = parent_segment._t_start
+                if not ignore_times:
+                    assert segment_t_start is None, ("ConcatenateSegmentSorting does not handle Sorting.t_start. "
+                                                     "Set time information only in the sortings' assigned recordings, "
+                                                     "or use ignore_times=True to ignore time information.")
+                # Pull num samples for each segment
+                if sorting.has_recording():
+                    segment_num_samples = sorting.get_num_samples(segment_index=segment_i)
+                else:
+                    assert segment_i == 0  # Already checked: sortings should be monosegment if they have no rec
+                    segment_num_samples = total_samples_list[sorting_i]
+                # Check consistency between num samples and spike frames
+                for unit_id in sorting.unit_ids:
+                    unit_segment_spikes = parent_segment.get_unit_spike_train(
+                        unit_id=unit_id, start_frame=None, end_frame=None,
+                    )
+                    if any([spike_frame >= segment_num_samples for spike_frame in unit_segment_spikes]):
+                        raise ValueError(
+                            "Sortings' spike frames exceed the provided number of samples for some segment. "
+                            "If the sortings have registered recordings, you can remove these excess "
+                            "spikes with `spikeinterface.curation.remove_excess_spikes(sorting, sorting._recording)`. "
+                            "Otherwise, the `total_num_samples` argument may contain invalid values."
+                        )
+                parent_segments.append(parent_segment)
+                parent_num_samples.append(segment_num_samples)
+        self.parent_num_samples = parent_num_samples
+
+        # Add a single Concatenated segment
+        sorting_seg = ProxyConcatenateSortingSegment(parent_segments, parent_num_samples,
+                                                     one_sorting.get_sampling_frequency())
+        self.add_sorting_segment(sorting_seg)
+
+        # Assign concatenated recording if possible
+        if all_has_recording:
+            self.register_recording(
+                concatenate_recordings([s._recording for s in sorting_list], ignore_times=ignore_times)
+            )
+
+        self._kwargs = {'sorting_list': [sorting.to_dict() for sorting in sorting_list],
+                        'ignore_times': ignore_times}
+
+    def get_num_samples(self, segment_index=None):
```

Comment:
yes. Are you referring to a line of code in particular?

Reply:
yes, what I'm saying is that if segment_index>1 it should fail, see below

Reply:
I think that the current behavior is the same as with Recording.get_num_samples:

`segment_index=None` works (`check_segment_index` returns 0 since it's single-segment)
`segment_index=0` works
`segment_index>0` fails with IndexError  on this line: https://github.com/SpikeInterface/spikeinterface/blob/8b94ac1d9b01c54250470d69394775c3113f454e/spikeinterface/core/segmentutils.py#L427